### ISSUE TITLE
Request to add time-window choices even if CGM data is (very) stale

### ISSUE THREAD

**Issue by mhaeberli** [nightscout/cgm-remote-monitor#1867]
If CGM data is current or recent, the Nightscout web view shows a choice, near the top-left, of 3 / 6 / 12 / 24 hour time windows.

If the CGM data is very stale, this choice disappears.

It's probably off-label use of Nightscout :-) , but we sometimes have RileyLink or other pump data, relatively recent, including BGs and some treatments, but very stale CGM data.  It would be very nice if the time-window choice were still present, even when CGM data is so stale.
##

**Comment by Mendanbar** [comment]
Just wanted to agree to this request.  We just had a situation where we've been off CGM for a few days (waiting on late sensor shipment), and when the data is very stale, the interface is very hard to use/read.  I'm also getting all kinds of bundle.js errors in the console (see attached).  Perhaps nightscout isn't failing gracefully in the case where there is no cgm data to display?
![bundle errors](https://cloud.githubusercontent.com/assets/10580609/20865525/efb43088-b9c9-11e6-93b1-c31e4f5c2c72.png)

**Comment by Kdisimone** [comment]
@mhaeberli when you say "very stale", how old do you mean?  My experience has been that if there is still data in the last 24-48 hours, that time selection still appears.  Since the display is only max 24 hours for the top chart, and 48 hours for bottom chart...this wouldn't appear to be an issue so long as you still have data within the last 48 hours.  You can use the bottom chart to scroll back in time and see any of the last 48 hours regardless of whether the 3/6/12/24 pills would be showing.

For example, below there are large swaths of time where no BG data is recorded, but the pills remain.  AND the user can grab the bottom graph to slide (or click to move wholesale) back in time to view data. 

<img width="1351" alt="screen shot 2018-02-21 at 11 25 01 am" src="https://user-images.githubusercontent.com/12402160/36500718-e75cdf28-16f9-11e8-97a5-6a6f835ed809.png">

I think once you get beyond 48 hours, you probably should be looking at other viewing screens, such as the reports day-to-day view.

**Comment by Kdisimone** [comment]
(@mhaeberli and if this issue has resolved for you, can you close this issue?  i realize this is quite dated now)

**Comment by Mendanbar** [comment]
I believe the request is still valid, but I'm unable to test at the moment.  The use case where it failed for us was as follows:
1. We ran out of Dexcom sensors, and had to wait several (at least 3, if memory serves) days to get new ones
2. in the mean time, we wanted to manually enter BG values using the care portal menu (plus button) in the nightscout website itself so we at least had a log of readings

What we saw, is that when there is no data from the CGM for several days, the interface becomes broken (see my screenshot above) and the care portal entries are no longer visible, even if they occur within the 24HR window supported by the interface.  Moreover, the selection box for the time range also disappears.

I'm trying to piece this together from memory... we've since learned our lesson and plan way ahead to avoid sensor shortages.

TLDR: Care Portal data can't be seen, and time window can't be selected, if CGM data is more than 24hrs(maybe 48?) stale

**Comment by Kdisimone** [comment]
@Mendanbar Your comment is actually along similar lines as the discussion thread in this comment https://github.com/nightscout/cgm-remote-monitor/issues/1532 

if you read the end message by @sulkaharo, the presence of CGM data via the entries collection (not just through treatments collection in careportal) is essential to the NS structure.  I think this issue could be tracked under that issue and closed here...as the issue is the same.

**Comment by mhaeberli** [comment]
Katie,
I concur, from a description standpoint.
I'm happy to let this one close out.
It _would_ be nice to be able to render even if no recent CGM entries and,
say zero or a few carelink entries.
But...
Hope all is well!
Best,
Martin

**Comment by PieterGit** [comment]
Closing this, and keeping https://github.com/nightscout/cgm-remote-monitor/issues/1532 open.
